modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobContainer.java

```diff
@@ -899,11 +899,11 @@ public void compareAndExchangeRegister(
             logger.trace(() -> Strings.format("[%s]: compareAndExchangeRegister failed", key), e);
             if (e instanceof AmazonS3Exception amazonS3Exception
                 && (amazonS3Exception.getStatusCode() == 404
-                    || amazonS3Exception.getStatusCode() == 0 && "NoSuchUpload".equals(amazonS3Exception.getErrorCode()))) {
+                    || amazonS3Exception.getStatusCode() == 200 && "NoSuchUpload".equals(amazonS3Exception.getErrorCode()))) {
                 // An uncaught 404 means that our multipart upload was aborted by a concurrent operation before we could complete it.
                 // Also (rarely) S3 can start processing the request during a concurrent abort and this can result in a 200 OK with an
-                // <Error><Code>NoSuchUpload</Code>... in the response, which the SDK translates to status code 0. Either way, this means
-                // that our write encountered contention:
+                // <Error><Code>NoSuchUpload</Code>... in the response, which the SDK translates to status code 200 since 1.12.691
+                // (0 for prior versions). Either way, this means that our write encountered contention:
```

Comment:
No need to talk about prior versions (or translation) here any more IMO: 

```suggestion
                // <Error><Code>NoSuchUpload</Code>... in the response. Either way, this means that our write encountered contention:
```